docs/docs/production-deployment.md

```diff
@@ -0,0 +1,161 @@
+
+Each instance of Pomerium runs in all-in-one mode unless specified to run as a specific component by the `services` key. See [All-In-One vs Split Service mode](/reference/readme.md#all-in-one-vs-split-service-mode) for more details.
+
+It's important to note that any production deployment with more than one instance of Pomerium (in any combination of modes) should be configured to use Redis as the [`databroker_storage_type`](/reference/readme.md#data-broker-storage-type). See [Data Storage - Backends](/docs/topics/data-storage.md#backends) for more details.
+
+### All in One
+
+In smaller deployments or while testing, it may be desirable to run in "all-in-one" mode. This reduces the resource footprint and simplifies DNS configuration. All URLs point at the same pomerium service instance.
+
+### Discrete Services
+
+In larger footprints, it is recommended to run Pomerium as a collection of discrete service clusters. This limits blast radius in the event of vulnerabilities and allows for per-service [scaling](#scaling) and monitoring.
+
+Please also see [Architecture](/docs/architecture.md) for information on component interactions.
+
+## Scaling
+
+In split service mode, you have the opportunity to scale the components of Pomerium independently.
+
+All of Pomerium's components are designed to be stateless, and may all be scaled horizontally or vertically. In general, horizontal scaling is recommended. Vertical scaling will lead to diminished returns after ~8 vCPUs.
+
+The Databroker service, which is responsible for session and identity related data, must be [configured for external persistence](/docs/topics/data-storage.md) to be fully stateless.
+
+Pomerium's individual components can be divided into two categories; the data plane and control plane. Regardless of which mode you run Pomerium in, we strongly recommend multiple instances of each service for fault tolerance.
+
+::: tip
+Our [Helm deployment](/docs/k8s/helm.md) supports [Horizontal Pod Autoscaling](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/).
+:::
+
+### Data Plane
+
+#### Proxy
+
+The Proxy service, as the name implies, is responsible for proxying all user traffic, in addition to performing checks to the Authorization service. The proxy is directly in path for user traffic.
+
+Proxy will need resources scaled in conjunction with request count and may need average request size accounted for. The heavier your user traffic, the more resources the Proxy service should have provisioned.
+
+#### Authorize
+
+The Authorize service is responsible for policy checks during requests. It is in the hot path for user requests but does not directly handle user traffic.
+
+Authorize will need resources scaled in conjunction with request count. Request size and type should be of a constant complexity. In most environments, Authorize and Proxy will scale linearly with request volume (user traffic).
```

Comment:
no actual change, but "Request size and type should be of a constant complexity." doesn't make sense to me. If I'm just not understanding lingo, ignore.